### PR TITLE
DATAUP-389 - add a job requirements resolver, pt 1

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

First step of adding a class to determine the requirements of a job from 3 sources - user input, the catalog record for the method, and the EE2 configuration file.

This PR adds a method to normalize job requirements given in `dict` format - e.g. parse strings to ints, etc. This is particularly useful for the catalog and config file sources since the former can present as a CSV string and the latter is an ini file, so all values are strings.

Expect 4 total PRs to complete the requirements resolver class.

# Jira Ticket / Github Issue #
- [X] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [X] Tests pass in Github Actions and locally 
- [X] Changes available by spinning up a local test suite

# Dev Checklist:

- [X] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - 3k warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [X] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the Github Actions build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: test/tests_for_utils/job_requirements_resolver_test.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+                "request_disk": None,
+                "client_group": None,
+                "client_group_regex": None,
+                "debug_mode": None,
+                "expect_noop": " fooo  ",
+            },
+            "source",
+        )
+        == {}
+    )
```
Comment (MrCreosote) [inline test/tests_for_utils/job_requirements_resolver_test.py:29]: so much more readable

**Comment by MrCreosote** [comment]
Looks like I forgot to take out an import when chopping up my changes into different PRs, but it'll get used later, so I propose just letting it sit.

File: lib/execution_engine2/utils/job_requirements_resolver.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+
+def _check_memory(memory, source):
+    if type(memory) == int:
+        return memory
+    memory2 = _string_request(memory, "memory request", source)
+    if memory2.endswith("M"):
+        memory2 = memory2[:-1]
+    elif memory2.endswith("MB"):
+        memory2 = memory2[:-2]
```
Comment (ialarmedalien) [inline lib/execution_engine2/utils/job_requirements_resolver.py:60]: are those the only options?

File: lib/execution_engine2/utils/job_requirements_resolver.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        {DEBUG_MODE}: parses to a boolean. The strings true and false are parsed to booleans,
+            case-insensitive. Ints are parsed directly to booleans.
+
+        reqs - the job requirements
+        source - the source of the job requirements, e.g. catalog, user, etc.
+        require_all_resources - True to throw an error if all four keys resources keys
+            ({CLIENT_GROUP}, {REQUEST_CPUS}, {REQUEST_MEMORY}, {REQUEST_DISK}) aren't present
+            with valid values.
+
+        Returns a a new dictionary with the altered keys. If any key is not present no action is
```
Comment (ialarmedalien) [inline lib/execution_engine2/utils/job_requirements_resolver.py:135]: double a

File: test/tests_for_utils/job_requirements_resolver_test.py (hunk 1 above)
Comment (ialarmedalien) [reply test/tests_for_utils/job_requirements_resolver_test.py:29]: it looks very artistic?

File: lib/execution_engine2/utils/job_requirements_resolver.py (hunk 2 above)
Comment (ialarmedalien) [reply lib/execution_engine2/utils/job_requirements_resolver.py:60]: (i.e. no GB / TB / KB / etc.)

Comment (MrCreosote) [reply lib/execution_engine2/utils/job_requirements_resolver.py:60]: Currently yes, the three sources of job requirements are:
- the run as concierge method, which takes ints
- the catalog, which currently only specifies numbers as strings with no units
- https://github.com/kbase/execution_engine2/blob/develop/deploy.cfg, which is `M` (and `GB` for disk space)

File: lib/execution_engine2/utils/job_requirements_resolver.py (hunk 3 above)
Comment (MrCreosote) [reply lib/execution_engine2/utils/job_requirements_resolver.py]: Fixed

**Review by ialarmedalien — APPROVED** [review]
👍

**Pull request merged**